### ISSUE TITLE
[Python] Array.windowed raises IndexError

### ISSUE THREAD

**Issue by bergmannjg** [fable-compiler/Fable#2815]
### Description

Array.windowed raises an IndexError with fable-py, the error has no effect with fable js

### Repro code

```f#
let arr = [|1;2;3|]
let w = arr |> Array.windowed 2
printfn "%A %A" arr w
```

### Expected and actual results

expected
```
array('i', [1, 2, 3]) [array('i', [1, 2]), array('i', [2, 3])]
```
actual 
```
IndexError: list assignment index out of range
```

i think the expression
```f#
source.Length - windowSize
```
in
https://github.com/fable-compiler/Fable/blob/3a06765ee30cc1289d785bddfd49bb432f53143c/src/fable-library/Array.fs#L901

should be changed to
```f#
source.Length - windowSize + 1
```

like [here](https://github.com/dotnet/fsharp/blob/2bdae3516274907a064d02641c75740653694a76/src/fsharp/FSharp.Core/array.fs#L783)

### Related information

* Fable version: 4.0.0-alpha-032
* Operating system; Ubuntu 20.04

**Comment by alfonsogarciacaro** [comment]
Thanks a lot for the detailed report @bergmannjg. Could you please have a look @dbrattli?

**Comment by dbrattli** [comment]
Yes, will do. I will btw be travelling this week, so no ETA for now.

**Comment by dbrattli** [comment]
Yes, this looks to be a bug with Fable and the F# code and not with Python. JavaScript survives since it will just resize the array automatically when you reference outside the pre-allocated size. Thanks for the detailed report @bergmannjg. It's spot on.

**Comment by alfonsogarciacaro** [comment]
Fixed by #2821
